### PR TITLE
Builds doc

### PR DESCRIPTION
**Pull request by mattmanning**
Explains what builds are and how they work.
Includes info about build export/import.

## Release Playbook
- [x] Rebase against master
- [x] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

**Comment by mattmanning** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Rq0dIryA&#x2F;689-builds-doc"> Builds doc</a>

File: _deployment/builds.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+title: "Builds"
+order: 250
+---
+
+A Build artifact is created when you deploy or build your application.
```
Comment (ianmalott) [inline _deployment/builds.md]: I feel like the term "artifact" here could use a bit more explanation. What can I do with a build artifact? What does Convox do with build artifacts by default? Is a "build" just shorthand for a "build artifact"? Could potentially tie the term into the "Moving Builds" section below...?

File: _deployment/builds.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- The CLI uploads the tarball to your Rack
+- The Rack extracts the tarball and reads `docker-compose.yml`
+- Docker images are built or pulled as specified by `docker-compose.yml`
+- The images are tagged and pushed into your Rack's private Amazon ECR registry
+- Build metadata is saved to the Rack
+- A new [release](/docs/releases) is created from the build and its metadata is saved too
+
+The newly created release will not be promoted (made active) until you run `convox releases promote <release ID>`.
+
+If you'd like to build your app and promote the release in a single step, you can run `convox deploy` rather than `convox build`.
```
Comment (ianmalott) [inline _deployment/builds.md]: 16-28: big fan of this breakdown

Comment (mattmanning) [inline _deployment/builds.md]: Thanks!

**Pull request merged**